helm/config.go

```diff
@@ -153,19 +154,56 @@ type Node interface {
 	write(enc *Encoder, prefix string)
 }
 
+// NewNode creates a new scalar node.
+func NewNode(value interface{}, modifiers ...NodeModifier) Node {
+	var node Node
+	if value == nil {
+		node = &Scalar{value: "~"}
+	} else if _, ok := value.(Node); ok {
+		node = value.(Node)
+	} else {
+		switch reflect.TypeOf(value).Kind() {
+		case reflect.Map:
+			valueOf := reflect.ValueOf(value)
+			mapping := NewEmptyMapping()
+			for _, key := range valueOf.MapKeys() {
+				mapping.Add(key.Interface().(string), valueOf.MapIndex(key).Interface())
+			}
+			node = mapping.Sort()
+		case reflect.Slice:
+			valueOf := reflect.ValueOf(value)
+			list := NewEmptyList()
+			for i := 0; i < valueOf.Len(); i++ {
+				list.Add(valueOf.Index(i).Interface())
+			}
+			node = list
+		case reflect.Bool:
+			node = &Scalar{value: strconv.FormatBool(value.(bool))}
+		case reflect.Int:
+			node = &Scalar{value: strconv.FormatInt(int64(value.(int)), 10)}
+		case reflect.Float64:
+			node = &Scalar{value: strconv.FormatFloat(value.(float64), 'G', -1, 64)}
+		case reflect.String:
+			str := value.(string)
+			// don't quote values that are templates from start to end
+			if !(strings.HasPrefix(str, "{{") && strings.HasSuffix(str, "}}")) {
+				str = strconv.Quote(str)
+			}
+			node = &Scalar{value: str}
+		default:
+			panic(fmt.Sprintf("Unexpected type: %s", reflect.TypeOf(value).Kind()))
+		}
+	}
+	node.Set(modifiers...)
+	return node
+}
+
 // Scalar represents a scalar value inside a list or mapping.
 type Scalar struct {
 	sharedFields
 	value string
 }
 
-// NewScalar creates a scalar node and initializes shared fields.
-func NewScalar(value string, modifiers ...NodeModifier) *Scalar {
-	scalar := &Scalar{value: value}
-	scalar.Set(modifiers...)
-	return scalar
-}
-
 // SetValue updates the value of a scalar node.
 func (scalar *Scalar) SetValue(value string) {
```

Comment:
`SetValue()` should take an `interface{}` argument instead of a `string`, so the user doesn't have to deal with quoting.

Reply:
https://github.com/SUSE/fissile/pull/282/commits/0ed98d295f2ada6bfa5590e0688a379f9f61d580